external-api/src/v1/controllers/party-db.controller.ts

```diff
@@ -32,3 +34,30 @@ export const lookup = async (req: Request, res: Response) => {
 
   return res.status(status).send(data);
 };
+
+export const createParty = async (req: CustomExpressRequest<{ reqBody: { companyName: string } }>, res: Response) => {
+  const { partyDbCompanyRegistrationNumber: companyReg } = req.params;
+  const { companyName } = req.body;
+
+  if (!isValidCompanyRegistrationNumber(companyReg)) {
+    console.error('Invalid company registration number provided %s', companyReg);
+    return res.status(HttpStatusCode.BadRequest).send({ status: HttpStatusCode.BadRequest, data: 'Invalid company registration number' });
+  }
+
+  const response: { status: number; data: unknown } = await axios({
```

Comment:
the response needs a status (following the pattern elsewhere in dtfs) so that we can distinguish between a 404 and an actual failure when querying SF - we don't want to create customers unless we definitely have a 404